src/main/java/org/zalando/nakadi/controller/ExceptionHandling.java

```diff
@@ -85,4 +87,18 @@ private String generateErrorTraceId() {
         }
         throw exception.getException();
     }
+
+    @ExceptionHandler(RepositoryProblemException.class)
+    public ResponseEntity<Problem> handleRepositoryProblem(final RepositoryProblemException exception,
+                                                           final NativeWebRequest request) {
+        LOG.error("Repository problem occurred", exception);
+        return Responses.create(Response.Status.SERVICE_UNAVAILABLE, exception.getMessage(), request);
+    }
+
+    @ExceptionHandler(InconsistentStateException.class)
```

Comment:
I was considering two reasons to have this:
1) I want to log the exception explicitly.
2) I want to expose to user only the exception message, not the stacktrace.